pymeasure/instruments/agilent/agilentB1500.py

```
@@ -914,22 +903,22 @@ def query_meas_ranges(self):
     :param str name: Name of the SMU
+    :param int slot: Slot number of the SMU
     """
 
-    def __init__(self, parent, channel, smu_type, name, **kwargs):
-        # to allow garbage collection for cyclic references
-        self._b1500 = weakref.proxy(parent)
-        channel = strict_discrete_set(channel, range(1, 11))
-        self.channel = channel
-        smu_type = strict_discrete_set(
-            smu_type,
+    def __init__(self, parent, index, type, name, slot, **kwargs):
```

Ruff (0.15.6)

[error] Function argument `type` is shadowing a Python builtin

(A002)